test/mlir/krnl/krnl_random_normal_lowering.mlir

```diff
@@ -54,3 +54,74 @@ func @test_random_normal_lowering() -> memref<3x4x5xf32> {
   // CHECK: llvm.call @get_random_normal_value_f32([[ALIGNED_TENSOR_MEMORY]], [[ALL_VALUES]], [[MEAN]], [[SCALE]], [[SEED]]) : (!llvm.ptr<f32>, i64, f32, f32, f32) -> ()
   // CHECK: llvm.return [[OUTPUT_TENSOR]] : !llvm.struct<(ptr<f32>, ptr<f32>, i64, array<3 x i64>, array<3 x i64>)>
 }
+
+// -----
+
+func @test_random_normal_dynamic_lowering(%arg0: memref<3x4x?x?xf32>) -> memref<3x4x?x?xf32> {
+  %c2 = arith.constant 2 : index
+  %0 = memref.dim %arg0, %c2 : memref<3x4x?x?xf32>
+  %c3 = arith.constant 3 : index
+  %1 = memref.dim %arg0, %c3 : memref<3x4x?x?xf32>
+  %2 = memref.alloc(%0, %1) {alignment = 16 : i64} : memref<3x4x?x?xf32>
+  %c12 = arith.constant 12 : index
+  %c2_0 = arith.constant 2 : index
+  %3 = memref.dim %arg0, %c2_0 : memref<3x4x?x?xf32>
+  %4 = arith.muli %c12, %3 : index
+  %c3_1 = arith.constant 3 : index
+  %5 = memref.dim %arg0, %c3_1 : memref<3x4x?x?xf32>
+  %6 = arith.muli %4, %5 : index
+  %cst = arith.constant 0.000000e+00 : f32
+  %cst_2 = arith.constant 1.000000e+00 : f32
+  %cst_3 = arith.constant 2.000000e+00 : f32
+  "krnl.random_normal"(%2, %6, %cst, %cst_2, %cst_3) : (memref<3x4x?x?xf32>, index, f32, f32, f32) -> ()
+  return %2 : memref<3x4x?x?xf32>
+
+  // CHECK: llvm.func @get_random_normal_value_f32(!llvm.ptr<f32>, i64, f32, f32, f32)
+  // CHECK: llvm.func @malloc(i64) -> !llvm.ptr<i8>
+  // CHECK: llvm.func @test_random_normal_dynamic_lowering(%arg0: !llvm.ptr<f32>, %arg1: !llvm.ptr<f32>, %arg2: i64, %arg3: i64, %arg4: i64, %arg5: i64, %arg6: i64, %arg7: i64, %arg8: i64, %arg9: i64, %arg10: i64) -> !llvm.struct<(ptr<f32>, ptr<f32>, i64, array<4 x i64>, array<4 x i64>)> {
+  // CHECK: [[C3:%.+]] = llvm.mlir.constant(3 : index) : i64
+  // CHECK: [[C4:%.+]] = llvm.mlir.constant(4 : index) : i64
+  // CHECK: llvm.mlir.constant(1 : index) : i64
```

Comment:
missing assignment to a variable. In general is it possible to reduce the number of CHECKs to the minimum required to ensure the test contain correct lowering ?